src/renderers/dom/fiber/ReactDOMFiber.js

```diff
@@ -359,6 +364,65 @@ var DOMRenderer = ReactFiberReconciler({
     parentInstance.removeChild(child);
   },
 
+  canHydrateInstance(
+    instance: Instance | TextInstance,
+    type: string,
+    props: Props,
+  ): boolean {
+    return instance.nodeType === 1 && type === instance.nodeName.toLowerCase();
+  },
+
+  canHydrateTextInstance(instance: Instance | TextInstance): boolean {
+    return instance.nodeType === 3;
+  },
+
+  getNextHydratableSibling(
+    instance: Instance | TextInstance,
+  ): null | Instance | TextInstance {
+    let node = instance.nextSibling;
+    // Skip non-hydratable nodes.
+    /*
+    while (node && node.nodeType !== 1 && node.nodeType !== 3) {
+      node = node.nextSibling;
+    }
+    */
+    return (node: any);
+  },
+
+  getFirstHydratableChild(
+    parentInstance: Container | Instance,
+  ): null | Instance | TextInstance {
+    let next = parentInstance.firstChild;
+    // Skip non-hydratable nodes.
+    /*
+    while (next && next.nodeType !== 1 && next.nodeType !== 3) {
+      next = next.nextSibling;
+    }
+    */
+    return (next: any);
+  },
+
+  hydrateInstance(
+    instance: Instance,
+    type: string,
+    props: Props,
+    rootContainerInstance: Container,
+    internalInstanceHandle: Object,
+  ): void {
+    precacheFiberNode(internalInstanceHandle, instance);
+    // TODO: Possibly defer this until the commit phase where all the events
+    // get attached.
+    updateFiberProps(instance, props);
+    setInitialProperties(instance, type, props, rootContainerInstance);
```

Comment:
Will this be able to solve reviving input values (#7745) (whether now or after 16) or is solving that in ReactDOM core a non-goal?